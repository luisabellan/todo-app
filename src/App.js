import React, { useState, useEffect } from 'react';
import TodoList from './components/TodoList/TodoList';
import TodoForm from './components/TodoForm/TodoForm';
import Header from './components/Header/Header';
import Footer from './components/Footer/Footer';
import GoogleFontLoader from 'react-google-font-loader';

import './App.scss';

function App() {
  let todoItems = [];

  const [todos, setTodos] = useState(todoItems);

  useEffect(() => {
    if (localStorage.getItem('todos')) {
      setTodos(JSON.parse(localStorage.getItem('todos')));
    }
  }, []);

  // saves todo on local storage
  const saveData = (newTodos) => {

    localStorage.setItem('todos', JSON.stringify(newTodos));
  }
  // Class methods to update state
  const addItem = (e, item) => {
    e.preventDefault();

    const newItem = {
      name: item,
      id: Date.now(),
      completed: false
    };
    let todoItems = [...todos, newItem];

    /* 
        /\S+/.test(item) 
        if 0 or more whitespace characters only item will not be added       
    */

    if (/\S+/.test(item)) {
      setTodos(todoItems);
      saveData(todoItems);

    }
  };



  const toggleItem = (itemId) => {
    // console.log(itemId);

    setTodos(
      todos.map((item) => {
        // console.log(item);
        if (itemId === item.id) {
          return {
            ...item,
            completed: !item.completed
          };
        }

        return item;
      })
    );
  };
  const clearCompleted = (e) => {
    e.preventDefault();
    let todoItems = todos.filter((item) => item.completed === false);
    setTodos(todoItems);
    saveData(todoItems);
  };



  return (
    <div className="app-wrapper" data-testid="app">
      <GoogleFontLoader
        fonts={[
          {
            font: 'Indie Flower',
            weights: [800, '4000'],
          },
          {
            font: 'Indie Flower',
            weights: [800, 900],
          },
        ]}
        subsets={['cyrillic-ext', 'greek']}
      />
      <Header className="header" />
      <div className="main">
        <TodoForm
          className="todoform"
          addItem={addItem}
          clearCompleted={clearCompleted}
        />
        {/* {console.log(todos)} */}
        <TodoList style={{ fontFamily: 'Indie Flower, monospaced' }} data-testid="todolist" className="todolist" todos={todos} toggleItem={toggleItem} />
      </div >
      <Footer />
    </div >
  )
}

export default App;
