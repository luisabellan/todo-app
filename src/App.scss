body {
  display: flex;
  justify-content: center;
  justify-items: center;
  height: 100vh;
  overflow: hidden;

  .header {
    height: 15vh;
  }

  .main {
    height: 80vh;
  }

  .footer {
    display: flex;
    flex-direction: column;
    position: fixed;
    top: auto;
    .copyright {
      display: flex;
      justify-content: center;
      a {
        margin-left: 0.3rem;
        text-decoration: none;
        color: rgb(250, 74, 4);
      }
    }
  }
}
