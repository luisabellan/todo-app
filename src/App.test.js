import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import App from './App';

afterEach(cleanup);

it('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
test('renders h1 tag', () => {
  render(<App />);
  const title = screen.getByText(/todo app/i);
  expect(title).toBeInTheDocument();
});
