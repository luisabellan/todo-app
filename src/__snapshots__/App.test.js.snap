// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-wrapper"
  >
    <h2>
      Todo App
    </h2>
    <div
      class="main"
    >
      <form>
        <h2>
          New Task
        </h2>
        <input
          data-testid="todo-input"
          name="todoItem1"
          placeholder="New task"
          type="text"
          value=""
        />
        <button
          class="add-to-do-btn"
          style="font-family: Jomhuria,sans-serif; outline: none; font-size: 1.2rem;"
        >
          <span>
            Add
          </span>
        </button>
        <button
          class="clear-btn"
          style="font-family: Jomhuria,sans-serif; outline: none; font-size: 1.2rem;"
        >
          <span>
            Clean
          </span>
        </button>
      </form>
      <div
        class="todolist-wrapper"
      >
        <ul
          data-testid="todos-ul"
        >
          <div
            class="task"
          >
            <input
              name="item buy bread"
              type="checkbox"
              value="buy bread"
            />
            <span>
              buy bread
            </span>
          </div>
        </ul>
      </div>
    </div>
    <footer
      class="footer"
    >
      <span
        class="copyright"
      >
        Made with ðŸ’“ and â˜• by 
        <a
          href="https://luisabellan.com"
        >
          Luis Abellan
        </a>
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
